src/plugins/data/public/search/collectors/create_usage_collector.ts

```diff
@@ -20,22 +21,27 @@ export const createUsageCollector = (
     return application.currentAppId$.pipe(first()).toPromise();
   };
 
-  return {
-    trackQueryTimedOut: async () => {
-      const currentApp = await getCurrentApp();
-      return usageCollection?.reportUiCounter(
-        currentApp!,
-        METRIC_TYPE.LOADED,
-        SEARCH_EVENT_TYPE.QUERY_TIMED_OUT
-      );
-    },
-    trackQueriesCancelled: async () => {
+  const getCollector = (metricType: UiCounterMetricType, eventType: SEARCH_EVENT_TYPE) => {
+    return async () => {
       const currentApp = await getCurrentApp();
-      return usageCollection?.reportUiCounter(
-        currentApp!,
-        METRIC_TYPE.LOADED,
-        SEARCH_EVENT_TYPE.QUERIES_CANCELLED
-      );
-    },
+      return usageCollection?.reportUiCounter(currentApp!, metricType, eventType);
+    };
+  };
+
+  return {
+    trackQueryTimedOut: getCollector(METRIC_TYPE.LOADED, SEARCH_EVENT_TYPE.QUERY_TIMED_OUT),
```

Comment:
track amount of searches \ new sessions?

Reply:
Can we do this in a follow-up PR for server-side telemetry?